lib/cacheWrapper.js

```diff
@@ -1,76 +1,112 @@
+      policies[cachePolicy.segment] = new Catbox.Policy(
+        {
+          expiresIn: cachePolicy.expiresIn
+        },
+        cacheClient, cachePolicy.segment
+    );
+    }
+  });
 
   // fire up the client
-  cacheClient.start( function() {
-    deferred.resolve();
-  } );
+  cacheClient.start(function() {
+    redisWrapper.initialise(serverConfig)
+    .then(function() {
+      deferred.resolve();
+    })
+    .fail(function() {
+      deferred.resolve('redis client was not initialised');
+    });
+  });
   return deferred.promise;
 };
 
 /**
 * Finds a value for the given key in the given segment of the memory store.
 * @param {Object} deferred - A promise to manage flow.
 * @param {Object} options - Including segment and key.
-* @returns {null} Doesn't explicitly return, simply handles the promis passed in.
+* @returns {null} Doesn't explicitly return, simply handles the promise passed in.
 */
-cacheWrapper._retrieve = function( deferred, options ) {
-  if ( policies[options.segment] ) {
-    policies[options.segment].get( options.key, function( error, cached ) {
-      if ( error || !cached ) {
+cacheWrapper._retrieve = function(deferred, options) {
+  if (policies[options.segment]) {
+    policies[options.segment].get(options.key, function(error, cached) {
+      if (error || !cached) {
         // catbox (Boom) errors are junk, we standardise by returning proper js errors
-        deferred.reject( new Error( 'retrieve failed' ) );
+        deferred.reject(new Error('retrieve failed'));
       } else {
-        deferred.resolve( cached );
+        deferred.resolve(cached);
       }
-    } );
+    });
   } else {
-    deferred.reject( new Error( 'policy not found for segment' ) );
+    deferred.reject(new Error('policy not found for segment'));
   }
 };
 
 /**
 * Sets a value for the given key in the given segment of the memory store.
 * @param {Object} deferred - A promise to manage flow.
 * @param {Object} options - Including segment, key and value.
-* @returns {null} Doesn't explicitly return, simply handles the promis passed in.
+* @returns {null} Doesn't explicitly return, simply handles the promise passed in.
 */
-cacheWrapper._stash = function( deferred, options ) {
-  if ( policies[options.segment] ) {
-    policies[options.segment].set( options.key, options.value, null, function( error ) {
-      if ( error ) {
+cacheWrapper._stash = function(deferred, options) {
+  if (policies[options.segment]) {
+    policies[options.segment].set(options.key, options.value, null, function(error) {
+      if (error) {
         // catbox (Boom) errors are junk, we standardise by returning proper js errors
-        deferred.reject( new Error( 'stash failed' ) );
+        deferred.reject(new Error('stash failed'));
       } else {
         deferred.resolve();
       }
-    } );
+    });
+  } else {
+    deferred.reject(new Error('policy not found for segment'));
+  }
+};
+
+cacheWrapper._delete = function(options) {
```

Comment:
I think it would be more readable to use a map here, instead of each:

```
if (!policies[options.segment]) {
  return Q.reject(new Error('policy not found for segment'));
}

var dropPromises = options.keys.map(function(key) {
  var deferred = Q.defer();
  policies[options.segment].drop(key, function() {
    deferred.resolve();
  });
  return deferred;
});

return Q.allSettled(dropPromises);
```